server/filesystem/compress.go

```diff
@@ -29,12 +29,21 @@ import (
 // and the compressed file will be placed at that location named
 // `archive-{date}.tar.gz`.
 func (fs *Filesystem) CompressFiles(dir string, paths []string) (ufs.FileInfo, error) {
+	var validPaths []string
 	for _, file := range paths {
 		if err := fs.IsIgnored(path.Join(dir, file)); err != nil {
-			return nil, err
+			// file is in file denylist so skip it and continue to check the next files
+			continue
```

Comment:
If you ever have an if/else statement (or just if in this case) that does nothing when it matches it's a good sign that you're using the wrong condition.  In this case you should check if the file is not ignored, then add it to validPaths.

Minor nit that's just a best practice, otherwise looks good to me!

Reply:
yes ofc, I changed it

```go
	var validPaths []string
	for _, file := range paths {
		if err := fs.IsIgnored(path.Join(dir, file)); err == nil {
			validPaths = append(validPaths, file)
		}
	}
```